consensus/quorum/one-node-staked-vote.go

```diff
@@ -1,75 +1,177 @@
 package quorum
 
 import (
+	"encoding/json"
 	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/harmony-one/bls/ffi/go/bls"
+	"github.com/harmony-one/harmony/internal/utils"
 	"github.com/harmony-one/harmony/numeric"
 	"github.com/harmony-one/harmony/shard"
+	"github.com/harmony-one/harmony/staking/slash"
 )
 
 var (
-	twoThirds = numeric.NewDec(2).QuoInt64(3).Int
+	twoThirds = numeric.NewDec(2).QuoInt64(3)
+	hSentinel = numeric.ZeroDec()
 )
 
 type stakedVoter struct {
 	isActive, isHarmonyNode bool
+	earningAccount          common.Address
 	effective               numeric.Dec
 }
 
 type stakedVoteWeight struct {
 	SignatureReader
 	DependencyInjectionWriter
-	// EPOS based staking
-	validatorStakes            map[[shard.PublicKeySizeInBytes]byte]stakedVoter
-	totalEffectiveStakedAmount *big.Int
+	DependencyInjectionReader
+	slash.ThresholdDecider
+	validatorStakes map[[shard.PublicKeySizeInBytes]byte]stakedVoter
```

Comment:
"[shard.PublicKeySizeInBytes]byte" - Use shard.BlsPublicKey directly. Also for the rest of this PR.